JS/Lesson18/goksel/script.js

```diff
@@ -27,17 +36,27 @@ console.log('----------------------------');
 // 3. Write a function that takes an array of strings and concatenates them together adding spaces.
 // Input: ["Hello", "World", "!"]
 // Output: "Hello World !"
-let stringEx3 = ["Hello", "World", "!"];
+let stringEx3 = ["Hello", "World", "!", "I am", "web developer", "?"];
 
 function concatenateStrings(array) {
   let sentence = '';
   // TODO: if the next character is one of ?, '.', ',', etc. do not add a space before that character
   for (let i = 0; i < array.length; i++) {
-    sentence = sentence + ' ' + array[i];
+    const specialCharacter = ["?",".",",","!",";",":","'","-","_","="];
+    const nextArrayCharacter = array[i+1];
+    if(specialCharacter.includes(nextArray)){
```

Comment:
The solution does not work right now. It's always good to run the code before submitting it just so you 100% that everything works :) 
The issue is the following:
- on line 46 variable declared with name nextArrayCharacter
- on line 47 instead of using the declared above variable, a non-existing variable is used ('nextArray').

When you change a name of a variable, make sure to check all the places where it was used and update! 

Also, current code structure is a bit hard to read and also is not working very well as I don't get a space between "!"and "I'am". Try to always construct the function the way you would construct a sentence:
**"If current character is a special character, do not add a space.".**

```

function concatenateStrings(array) {
  let sentence = '';
  // TODO: if the next character is one of ?, '.', ',', etc. do not add a space before that character
  for (let i = 0; i < array.length; i++) {
    const specialCharacter = ["?",".",",","!",";",":","'","-","_","="];
    if(specialCharacter.includes(array[i])){ // if current character is a special character
      sentence += array[i]; // Do no add a space
    }else {
      sentence = sentence + " " + array[i];
    }
  }
  return sentence; 
};
```